Amplify/Categories/DataStore/Model/Temporal/TemporalFormat.swift

```diff
@@ -7,37 +7,23 @@
 
 import Foundation
 
-/// `Enum` that represent the different options for formatting ISO8601 values.
-/// Each implementation of a `TemporalSpec` is responsible for converting the enum value
-/// to a corresponding format string.
-public enum TemporalFormat: CaseIterable {
+public protocol TemporalSpecValidFormatRepresentable: Equatable {
```

Comment:
name change in public protocol is a breaking change right? Even though developers might not be using this, it goes against the semver versioning.

Reply:
This is a new protocol, so it's not a name change. This is the protocol that defines structure of formats. 

Your point is valid one though. `TemporalFormat` was a public enum and doesn't exist anymore after these changes. However, we don't provide any public APIs that make use of the `TemporalFormat` enum. It only provided any value in combination with a `TemporalSpec.Type` through the **internal** `getFormat(for:) -> String` API. 

Removing can't break behavior because it has no public behavior. I don't believe this constitutes a semver breaking change. If you still have concerns despite this information, we can leave the type and deprecate it.

Reply:
I am not sure whether it is being used by developers, I see that we had to update the test below to make it work. 

Also I see here https://github.com/aws-amplify/amplify-ios/issues/1048 that there are developers who might be using the format to get a specific date format? 

@lawmicha might have more context here.

Reply:
The changed test uses the format with an internal API.

That linked issue discusses using `TemporalFormat` in a fork of Amplify, not as part of the Amplify public API surface.

Reply:
Actually what I said a few comments up is incorrect. There is a public API that takes a `TemporalFormat` as part of the `TemporalSpec` protocol 😕.
This would only matter if the format was being created before being used as a function argument.

```swift
// ✅ this would be fine 
_ = Temporal.DateTime.iso8601FormattedString(format: .full)

// ❌ this would break
let format = TemporalFormat.full.dateTime
_  = Temporal.DateTime.iso8601FormattedString(format: format) 
```

I'm going to take a look restructuring format to not break in this case. Might be able to even drop the new associatedtype requirement in `TemporalSpec` as well 🤞